jetty-security/src/main/java/org/eclipse/jetty/security/Authenticator.java

```diff
@@ -117,7 +117,20 @@ interface AuthConfiguration
 
         IdentityService getIdentityService();
 
+        /**
+         * Should session ID be renewed on authentication.
+         * @return true if the session ID should be renewed on authentication
+         */
         boolean isSessionRenewedOnAuthentication();
+
+        /**
+         * Get the interval in seconds to set with {@link javax.servlet.http.HttpSession#setMaxInactiveInterval(int)}
+         * when a session is newly authenticated.
+         * @return An interval in seconds, which if greater than zero will be set as the
+         *        {@link javax.servlet.http.HttpSession#setMaxInactiveInterval(int) maxInactiveInterval}
+         *        on newly authenticated sessions.
+         */
+        int getSessionMaxInactiveIntervalOnAuthentication();
```

Comment:
```suggestion
        int getAuthenticatedSessionMaxInactiveInterval();
```

Reply:
I've kept this as is, because the "OnAuthentication" ties it to other existing methods that trigger on the same event.